components/view/index.php

```diff
@@ -49,11 +53,15 @@
     });
 
     $viewResolver->register('php', function () {
-        return new PhpEngine;
+        return App::getInstance()->make(PhpEngine::class);
```

Comment:
Does this work? The PhpEngine class has not been bound anywhere, so it shouldn't right?

Reply:
It does as the container will call `new` with dependency injection for unbound classes.
A reason why you should avoid `new` calls in general in packages but also in your own code to make it easier to override a class or always call a method on/after initialization.
It will also try to get all dependencies from container or initialize them the same way as the requested class.

Reply:
Oh, cool I did not know that.

Reply:
Shouldn't this be done with `return new CompilerEngine($bladeCompiler);` then as well?

Reply:
I've mainly tried to avoid using any features of Slim other than routing.

And I've tried to keep the build steps *super* clear so that folks who don't *absolutely* have to use the Illuminate container don't have to.

Without having read the code, I would prefer we avoid pulling anything out of the container unless absolutely necessary.

Reply:
OK, reading the code, I'd prefer we keep it as `new ` if possible.

> For the blade engine this would lead to a lot more changes as all the dependencies would have to be added to the container.

Does that mean the old PHPEngine didn't have all these dependencies but the new one does?

Thanks!

Reply:
The PhpEngine change was only as PhpStorm complained about missing construct call. Haven't checked if that changed somewhen or it never worked but hasn't been a problem as we use the blade engine only in the example.

For this component slim is "useless" as it's a single route - so a plain index.php would be the same as the current construct.
So if you want to keep it simple I would say that we. can remove slim here.

Regarding the container: even if it's already required? So this component doesn't work without container as the internals use the container.
So I agree that we shouldn't use it just for comfort - but if it's already required we could use its full potential!?

Reply:
I'd still like to avoid using its full potential, because I know folks may choose to use other forms of IOC/dependency injection, so I'd like what we're doing to be laid out in its full, ugly glory :)

Reply:
Long discussion - short solution: the PhpEngine is now fully removed 😂
It's not needed for blade and those who want to use PHP templates should use `include` 😉